pkg/frontend/querymiddleware/experimental_operators.go

```diff
@@ -0,0 +1,151 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package querymiddleware
+
+import (
+	"context"
+	"fmt"
+	"slices"
+
+	"github.com/go-kit/log"
+	"github.com/grafana/dskit/tenant"
+	"github.com/prometheus/prometheus/promql/parser"
+
+	apierror "github.com/grafana/mimir/pkg/api/error"
+	"github.com/grafana/mimir/pkg/frontend/querymiddleware/astmapper"
+)
+
+const (
+	allExperimentalOperators = "all"
```

Comment:
[nit] I'd name this `allExperimentalFeatures` to avoid confusion given "operators" has a particular meaning in the context of MQE

(similar feedback applies for other uses of "operator" in this file)